lib/mayaUsd/ufe/Utils.cpp

```diff
@@ -76,25 +85,95 @@ UsdStageWeakPtr getStage(const Ufe::Path& path) { return g_StageMap.stage(path);
+        // Note here that we're taking advantage of the fact that identifiers
+        // in SdfPaths must be C/Python identifiers; that is, they must *not*
+        // begin with a digit. This means that when we see a path component at
+        // the end of a USD path segment that does begin with a digit, we can
+        // be sure that it represents an instance index and not a prim or other
+        // USD entity.
+        pathString += TfStringPrintf("%c%d", separator, instanceIndex);
+    }
+
+    return Ufe::PathSegment(pathString, usdRuntimeId, separator);
+}
+
+Ufe::Path stripInstanceIndexFromUfePath(const Ufe::Path& path)
+{
+    if (path.empty()) {
+        return path;
+    }
+
+    // As with usdPathToUfePathSegment() above, we're taking advantage of the
+    // fact that identifiers in SdfPaths must be C/Python identifiers; that is,
+    // they must *not* begin with a digit. This means that when we see a path
+    // component at the end of a USD path segment that does begin with a digit,
+    // we can be sure that it represents an instance index and not a prim or
+    // other USD entity.
+    const std::string& tailComponentString = path.back().string();
+    const char&        firstChar = tailComponentString.front();
+    if (std::isdigit(static_cast<unsigned char>(firstChar))) {
+        return path.pop();
+    }
+
+    return path;
+}
+
 UsdPrim ufePathToPrim(const Ufe::Path& path)
 {
+    const Ufe::Path ufePrimPath = stripInstanceIndexFromUfePath(path);
+
     // Assume that there are only two segments in the path, the first a Maya
     // Dag path segment to the proxy shape, which identifies the stage, and
     // the second the USD segment.
     // When called we do not make any assumption on whether or not the
     // input path is valid.
-    const Ufe::Path::Segments& segments = path.getSegments();
-    if (!TF_VERIFY(segments.size() == 2, kIllegalUSDPath, path.string().c_str())) {
+    const Ufe::Path::Segments& segments = ufePrimPath.getSegments();
+    if (!TF_VERIFY(segments.size() == 2u, kIllegalUSDPath, path.string().c_str())) {
         return UsdPrim();
     }
 
     UsdPrim prim;
     if (auto stage = getStage(Ufe::Path(segments[0]))) {
-        prim = stage->GetPrimAtPath(SdfPath(segments[1].string()));
+        const SdfPath usdPath = SdfPath(segments[1].string());
+        prim = stage->GetPrimAtPath(usdPath.GetPrimPath());
     }
     return prim;
 }
 
+int ufePathToInstanceIndex(const Ufe::Path& path)
+{
+    int instanceIndex = UsdImagingDelegate::ALL_INSTANCES;
+
+    const UsdPrim usdPrim = ufePathToPrim(path);
+    if (!usdPrim || !usdPrim.IsA<UsdGeomPointInstancer>()) {
+        return instanceIndex;
+    }
+
+    // Once more as above in usdPathToUfePathSegment() and
+    // stripInstanceIndexFromUfePath(), a path component at the tail of the
+    // path that begins with a digit is assumed to represent an instance index.
+    const std::string& tailComponentString = path.back().string();
```

Comment:
Might want to factor out the test as an (inline) predicate in the unnamed namespace in this file, and use it in both places.

Reply:
Good idea. Done.

Reply:
Addressed in dde6e1e46df178891d348b4d645cb4395b7546ff and 5a2f62f2ba8779ac48a339e524f14315eec23fec.